File: 01-matrixmultiply-Java/matrixmultiply.java
// # Write the function matrixMultiply(m1, m2) that takes two 2d arrays 
// # (that we will consider to be matrices) and returns a new 2d array that 
// # is the result of multiplying the two matrices. Return [[]] if the 
// # two matrices cannot be multiplied for any reason.

class matrixmultiply {
	public int[][] fun_matrixmultiply(int[][] a, int[][] b){
		int[][] arr = {{}};
		if(a[0].length != b.length){
			return arr;
		}else{
			int[][] res = new int[a.length][b[0].length];
			for(int i = 0; i < a.length; i++){
				for(int j = 0; j < b[0].length; j++){
					res[i][j] = 0;
					for(int k = 0; k < b.length; k++) {
						res[i][j] = res[i][j] + (a[i][k] * b[k][j]);
					}
				}
			}
			return res;
		}
	}

	public static void main(String[] args) {
		
	}
}

File: 01-matrixmultiply-Python/matrixmultiply.py
# Write the function matrixMultiply(m1, m2) that takes two 2d lists 
# (that we will consider to be matrices) and returns a new 2d list that 
# is the result of multiplying the two matrices. Return None if the 
# two matrices cannot be multiplied for any reason.

def fun_matrixmultiply(m1, m2):
    if len(m1[0]) != len(m2):
        return None
    return [[sum(a * b for a, b in zip(A_row, B_col)) for B_col in zip(*B)] for A_row in A]